packages/scm/src/browser/history/scm-history-widget.tsx

```diff
@@ -14,44 +14,74 @@
+import { Event as TheiaEvent, DisposableCollection } from '@theia/core';
 import { OpenerService, open, StatefulWidget, SELECTED_CLASS, WidgetManager, ApplicationShell } from '@theia/core/lib/browser';
 import { CancellationTokenSource } from '@theia/core/lib/common/cancellation';
 import { Message } from '@phosphor/messaging';
 import { AutoSizer, List, ListRowRenderer, ListRowProps, InfiniteLoader, IndexRange, ScrollParams, CellMeasurerCache, CellMeasurer } from 'react-virtualized';
-import { GIT_RESOURCE_SCHEME } from '../git-resource';
 import URI from '@theia/core/lib/common/uri';
-import { GIT_HISTORY_ID, GIT_HISTORY_MAX_COUNT, GIT_HISTORY_LABEL } from './git-history-contribution';
-import { GitFileStatus, Git, GitFileChange, Repository } from '../../common';
+import { ScmService } from '../scm-service';
+import { SCM_HISTORY_ID, SCM_HISTORY_MAX_COUNT, SCM_HISTORY_LABEL } from './scm-history-contribution';
+import { ScmCommit, ScmFileChange } from '../scm-provider';
 import { FileSystem } from '@theia/filesystem/lib/common';
-import { GitDiffContribution } from '../diff/git-diff-contribution';
-import { ScmAvatarService } from '@theia/scm/lib/browser/scm-avatar-service';
-import { GitCommitDetailUri, GitCommitDetailOpenerOptions, GitCommitDetailOpenHandler } from './git-commit-detail-open-handler';
-import { GitCommitDetails } from './git-commit-detail-widget';
-import { GitNavigableListWidget } from '../git-navigable-list-widget';
-import { GitFileChangeNode } from '../git-file-change-node';
+import { ScmAvatarService } from '../scm-avatar-service';
+import { ScmNavigableListWidget } from '../scm-navigable-list-widget';
 import * as React from 'react';
 import { AlertMessage } from '@theia/core/lib/browser/widgets/alert-message';
 
-export interface GitCommitNode extends GitCommitDetails {
-    fileChanges?: GitFileChange[];
+export const ScmHistorySupport = Symbol('scm-history-support');
+export interface ScmHistorySupport {
+    getCommitHistory(options?: HistoryWidgetOptions): Promise<ScmCommit[]>;
+    readonly onDidChangeHistory: TheiaEvent<void>;
+}
+
+export interface ScmCommitNode {
+    commitDetails: ScmCommit;
+    authorAvatar: string;
+    fileChangeNodes?: ScmFileChangeNode[];
     expanded: boolean;
     selected: boolean;
 }
+export interface ScmFileChangeNode {
+    readonly fileChange: ScmFileChange;
+    readonly icon: string;
+    readonly label: string;
+    readonly description: string;
+    readonly caption?: string;
+    readonly extraIconClassName?: string;
+    readonly commitSha?: string;
+    selected?: boolean;
+}
+
+export interface HistoryWidgetOptions {
+    readonly range?: {
+        readonly toRevision?: string;
+        readonly fromRevision?: string;
+    };
+    readonly uri?: string;
+    readonly maxCount?: number;
+}
+
+export namespace ScmCommitNode {
+    // tslint:disable-next-line:no-any
+    export function is(node: any): node is ScmCommitNode {
```

Comment:
Could use `unknown` rather than `any` here, but you'd need to replace `!!node` below with `typeof node === 'object' && node !== null && …`.

Reply:
This seems to be the pattern used in Theia and this code was moved and edited from existing Theia code so I think I should leave it as is.  I often get pushback for unnecessary changes :-)